src/Languages/Editor/Test/Shell/TestShellBase.cs

```diff
@@ -39,6 +39,10 @@ public class TestShellBase {
             await UIThreadHelper.Instance.InvokeAsync(action);
         }
 
+        public async Task<TResult> DispatchOnMainThreadAsync<TResult>(Func<TResult> action, CancellationToken cancellationToken = new CancellationToken()) {
```

Comment:
Technically DispatchOnUIThread was supposed to be async (it used to be Dispatcher.BeginInvoke long time ago). Code we inherited actually expects DispatchOnUIThread  to be non-blocking.